tests/utils.py

```diff
@@ -94,7 +96,39 @@ def get_llama_cpp_model(model_name, caching=False, **kwargs):
     key = model_name + "_" + str(caching) + "_" + str(kwargs)
     if key not in llama_cpp_model_cache:
         llama_cpp_model_cache[key] = guidance.models.LlamaCpp(
-            model_name, caching=caching, **kwargs
+            model_name, **kwargs
         )
 
     return llama_cpp_model_cache[key]
+
+
+azure_guidance_model_cache = {}
+azure_guidance_defaults = {}
+
+
+def get_azure_guidance_model(model_name, caching=False, **kwargs):
+    """Get Azure Guidance LLM with model reuse."""
+
+    if (
+        model_name is None
+        or isinstance(model_name, str)
+        and len(model_name.strip()) == 0
+    ):
+        model_name = os.getenv("AZURE_GUIDANCE_URL", "")
+        if len(model_name.strip()) == 0:
+            pytest.skip("No Azure Guidance model found.")
```

Comment:
I'd rather throw an exception here. Tests which skip themselves if misconfigured are the stuff of nightmares (although less ones compared to some of the other issues we have).

Reply:
good idea.  Harsha wants to merge, so I'll merge it now and fix this in a bit.